### PR TITLE
v3.0.0 Blog Post

### PR DESCRIPTION
**Pull request by devlead**

### CODE REVIEW

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+- Improved build providers
+- Addin recommended Cake version set to 3.0
+
+### .NET 7 Support & C#11 Support
+
+Cake now fully supports running on .NET 7, and with this C# 11, which means you can take advantage of the latest framework, runtime, and language improvements.
+
+### TaskOf&lt;T&gt;
+
+For scripting a new `TaskOf<T>` method has been introduced to simplify when working with shared typed data contexts, where one only needs to specify the type parameter once and only get task methods that are relevant, example:
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:27]:
```suggestion
For scripting, a new `TaskOf<T>` method has been introduced to simplify when working with shared typed data contexts, where one only needs to specify the type parameter once and only get task methods that are relevant, example:
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        new [] { true, false },
+        (data, item, context) => context.Information("Item: {0}, Initialized: {1}.", item, data.Initialized)
+    );
+
+RunTarget("TaskOfT");
+```
+
+### Multiple target support
+
+Also for scripting a new `RunTargets(IEnumerable<string> targets)` method is now available it'll let you specify multiple targets as the entry point in Cake's dependency graph to execute, where dependencies and dependency order is still respected, for example calling script below with `dotnet cake --target="A" --target="B"`
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:53]:
```suggestion
Also for scripting, a new `RunTargets(IEnumerable<string> targets)` method is now available. It'll let you specify multiple targets as the entry point in Cake's dependency graph to execute, where dependencies and dependency order is still respected. For example calling script below with `dotnet cake --target="A" --target="B"`
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+B                             00:00:00.0006161
+--------------------------------------------------
+Total:                        00:00:00.0041088
+```
+
+### Breaking changes
+
+Being a major release we've taken the opportunity to remove obsolete aliases and members, i.e. DotNetCore* aliases were obsoleted and replaced by DotNet* aliases in Cake 2.0 and are now removed with 3.0.
+
+Cake itself will no longer run on .NET Core 3.1 or .NET 5, but Cake will continue to support the building of .NET Framework projects as well as projects targeting .NET 5.0 or older.
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:99]:
```suggestion
Cake itself will no longer run on .NET Core 3.1 or .NET 5, but Cake will continue to support the building of .NET Framework projects, as well as projects targeting .NET 5.0 or older. 
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+Total:                        00:00:00.0041088
+```
+
+### Breaking changes
+
+Being a major release we've taken the opportunity to remove obsolete aliases and members, i.e. DotNetCore* aliases were obsoleted and replaced by DotNet* aliases in Cake 2.0 and are now removed with 3.0.
+
+Cake itself will no longer run on .NET Core 3.1 or .NET 5, but Cake will continue to support the building of .NET Framework projects as well as projects targeting .NET 5.0 or older. 
+
+The supported platform matrix for Cake 2.0 will look like this:
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:101]:
```suggestion
The supported platform matrix for Cake 3.0.0 will look like this:
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+| Runner                           | .NET 7                                          | .NET 6                                          |
+| -------------------------------- |:-----------------------------------------------:|:-----------------------------------------------:|
+| [Cake .NET Tool]                 | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> |
+| [Cake Frosting]                  | <i class="fa fa-check" style="color:green"></i> | <i class="fa fa-check" style="color:green"></i> |
+
+### Improved build providers
+
+Cake 3.0 comes with a few improvements to build providers i.e.
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:110]:
```suggestion
Cake 3.0.0 comes with a few improvements to build providers i.e.
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    - `SetOutputParameter` Command
+    - Workflow commands
+- TeamCity
+    - Build status message
+    - Statistics
+- GitLab
+    - `SetEnvironmentVariable` Command
+    - CI runner tags improvements
+
+### Addin recommended Cake version set to 3.0
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:123]:
```suggestion
### Addin recommended Cake version set to 3.0.0
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+We've now set the recommended version of [Cake.Core](https://www.nuget.org/packages/Cake.Core) for addins to target to 3.0.
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:125]:
```suggestion
We've now set the recommended version of [Cake.Core](https://www.nuget.org/packages/Cake.Core) for addins to target to 3.0.0.
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+This won't break the build but you might see warnings like below when addins or modules are loaded.
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:127]:
```suggestion
This won't break the build, but you might see warnings like below when addins or modules are loaded.
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+This release includes several new features, improvements and bug fixes to both [Cake Scripting](/docs/running-builds/runners/dotnet-tool) and [Cake Frosting](/docs/running-builds/runners/cake-frosting) since the [Cake v2.3.0 release](/blog/2022/10/cake-v2.3.0-released)! 🚀 🍰
+
+### Highlights of this release
+
+- .NET 7 Support & C#11 Support
+- TaskOf&lt;T&gt;
+- Multiple target support
+- Breaking changes
+- Improved build providers
+- Addin recommended Cake version set to 3.0
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:19]:
```suggestion
- Addin recommended Cake version set to 3.0.0
```

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+A                             00:00:00.0028166
+C                             00:00:00.0006761
+B                             00:00:00.0006161
+--------------------------------------------------
+Total:                        00:00:00.0041088
+```
+
+### Breaking changes
+
+Being a major release we've taken the opportunity to remove obsolete aliases and members, i.e. DotNetCore* aliases were obsoleted and replaced by DotNet* aliases in Cake 2.0 and are now removed with 3.0.
```
Comment (gep13) [inline input/blog/2022-11-08-cake-v3.0.0-released.md:97]:
```suggestion
Being a major release we've taken the opportunity to remove obsolete aliases and members, i.e. DotNetCore* aliases were obsoleted and replaced by DotNet* aliases in Cake 2.0 and are now removed with 3.0.0.
```

**Review by gep13 — CHANGES_REQUESTED** [review]

**Review by bjorkstromm — COMMENTED** [review]
LGTM when @gep13's and @pascalberger's comments are resolved.

File: input/blog/2022-11-08-cake-v3.0.0-released.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+Version **3.0.0** of Cake has been released. Take it for a spin and give us feedback on our [discussion board](https://github.com/cake-build/cake/discussions/xxxx).
+
+This release includes several new features, improvements and bug fixes to both [Cake Scripting](/docs/running-builds/runners/dotnet-tool) and [Cake Frosting](/docs/running-builds/runners/cake-frosting) since the [Cake v2.3.0 release](/blog/2022/10/cake-v2.3.0-released)! 🚀 🍰
+
+### Highlights of this release
+
+- .NET 7 Support & C#11 Support
+- TaskOf&lt;T&gt;
+- Multiple target support
+- Breaking changes
```
[the remark this answers is not in the document]
Comment (devlead) [reply input/blog/2022-11-08-cake-v3.0.0-released.md:17]: I think breaking changes are a pretty significant part of this release, isn't it?

[the remark this answers is not in the document]
Comment (pascalberger) [reply input/blog/2022-11-08-cake-v3.0.0-released.md:17]: Yes, breaking changes are important, but they are not a feature. Therefore my suggestion to remove Breaking Changes from this list and have a list of features, but instead make Breaking Changes the next section.

[the remark this answers is not in the document]
Comment (devlead) [reply input/blog/2022-11-08-cake-v3.0.0-released.md]: removed

**Review by pascalberger — APPROVED** [review]
LGTM

**Pull request merged**